### ISSUE TITLE
Add ignoreKeywords: [] option to value-keyword-case

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1082]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Some animation name may have names with camel case, example in https://daneden.github.io/animate.css/

**Comment by jeddy3** [comment]
Would you be better turning the rule off if you're going to be using camelCase to name your animations, especially as the [built in timing-function keywords](https://drafts.csswg.org/css-transitions-1/#single-transition-timing-function) are hyphenated?

**Comment by evilebottnawi** [comment]
@jeddy3 sometimes you may use keyword from other postcss plugins, everything is possible, it would be a good idea to give the possibility to ignore some keywords.

**Comment by davidtheclark** [comment]
Yeah, an `ignore` option sounds good.

**Comment by evilebottnawi** [comment]
@davidtheclark set `help wanted`, `enhancement`?

**Comment by jeddy3** [comment]
> sometimes you may use keyword from other postcss plugins

Yep, I realised that shortly after I wrote the comment. A user might be *consuming* 3rd-party animation names in their source code that they have no control over.

The option [should be](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#determine-options) `ignoreKeywords: []`.

**Comment by alienlebarge** [comment]
Another good use case of an ignore option is fonts.

```shell
src/assets/foehn/styles/_settings.fontface.css
17:9	⚠  Expected "Palatino" to be "palatino" (value-keyword-case) [stylelint]
18:9	⚠  Expected "Palladio" to be "palladio" (value-keyword-case) [stylelint]
21:9	⚠  Expected "Baskerville" to be "baskerville" (value-keyword-case) [stylelint]
25:9	⚠  Expected "Garamond" to be "garamond" (value-keyword-case) [stylelint]
31:9	⚠  Expected "Georgia" to be "georgia" (value-keyword-case) [stylelint]
58:31	⚠  Expected "Consolas" to be "consolas" (value-keyword-case) [stylelint]
60:9	⚠  Expected "Menlo" to be "menlo" (value-keyword-case) [stylelint]
61:9	⚠  Expected "Courier" to be "courier" (value-keyword-case) [stylelint]
```

**Comment by evilebottnawi** [comment]
@jeddy3 @davidtheclark what do you think we should ignore all `font names` as default or add options for this?

**Comment by davidtheclark** [comment]
Yes, I think all font names could be ignored.

**Comment by alienlebarge** [comment]
Will it not be hard to ignore all the font names by default ?
How do you do that?

A font name may be after *font-family* as well as set in a variable.  

Le 21 avril 2016 à 16:20:16, David Clark (notifications@github.com(mailto:notifications@github.com)) a écrit:

**Comment by evilebottnawi** [comment]
@alienlebarge i create new issue about this problem, here i add only `ignoreKeyword` option

**Comment by jeddy3** [comment]
Lets keep it here for a bit as I think we've stumbled into something that might make this option redundant.

**Comment by jeddy3** [comment]
As @alienlebarge point out... won't ignoring all font names be hard?

The only solution I can think of, and perhaps what we should of done from the start, is make this rule applicable to only *known* keywords from the specs.

Animation names and font names will then be ignored without the need for this option.

**Comment by evilebottnawi** [comment]
@jeddy3 good point at the expense of animation name, https://github.com/stylelint/stylelint/issues/1101

**Comment by alienlebarge** [comment]
What if a font has the same name than a keyword. I can easily imagine a font named *Flex*.

There's almost no chance but there's one chance.

**Comment by evilebottnawi** [comment]
@alienlebarge we ignore property not keywords

**Comment by jeddy3** [comment]
> @alienlebarge we ignore property not keywords

Perhaps we needn't be as blunt as that. According to [the specs](https://drafts.csswg.org/css-values/#custom-idents) properties that accept *custom-idents* must do in a "positionally unambiguous" way. Therefore, I think, we could identify which properties accept *custom-idents* and in which value position, and then *only* ignore keywords found in those positions.

Off hand, I can think of the following properties that accept *custom-idents*:

- `animation`
- `animation-timing-function`
- `animation-name`
- `will-change`
- `font`
- `font-family`
- `counter-increment`
- `grid-row`
- `grid-column`
- `grid-area`
- `list-style-type`

I think adding this feature would make your [recent PR](https://github.com/stylelint/stylelint/pull/1102) redundant and so I'm going to wait until we resolve this issue before review the PR.

**Comment by evilebottnawi** [comment]
@jeddy3 Duplicate it https://github.com/stylelint/stylelint/pull/1102, tomorrow i have time for this

**Comment by evilebottnawi** [comment]
@jeddy3 sorru wrong url https://github.com/stylelint/stylelint/issues/1101

**Comment by jeddy3** [comment]
Ok, perhaps this option is *complementary* to the ignoring *custom-ident* slots. I'll review your PR.

**Comment by evilebottnawi** [comment]
@jeddy3 yep only for custom keywords.
